src/libcore/tests/mem.rs

```diff
@@ -47,6 +47,36 @@ fn size_of_val_basic() {
     assert_eq!(size_of_val(&1u64), 8);
 }
 
+#[test]
+fn size_of_val_const() {
+    macro_rules! ez_byte_string {
+        ($name: ident, $string: expr) => {
+            static $name: [u8; size_of_val($string)] = *$string;
+        }
+    }
+
+    ez_byte_string!(OWO, b"what's this?");
+    ez_byte_string!(EMPTY, b"");
+
+    assert_eq!(&OWO, b"what's this?");
+    assert_eq!(&EMPTY, b"");
+
+    // ~ one day ~
+    // static SIZE_OF_OWO_SLICE: usize = size_of_val(&OWO[..]);
```

Comment:
you should be able to write

```rust
static FOO: &[u8] = &OWO;
static BAR: usize = size_of_val(FOO);
```

because coercions don't need method calls like slice indexing does